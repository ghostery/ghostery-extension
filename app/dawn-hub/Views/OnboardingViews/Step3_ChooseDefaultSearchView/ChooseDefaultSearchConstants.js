/**
 * Choose Default Search constants for use by the Dawn onboarding hub
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2021 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

export const SET_DEFAULT_SEARCH = 'SET_DEFAULT_SEARCH';
export const SEARCH_GHOSTERY = 'Ghostery';
export const SEARCH_BING = 'Bing';
export const SEARCH_YAHOO = 'Yahoo';
export const SEARCH_STARTPAGE = 'StartPage';
export const SEARCH_DUCKDUCK_GO = 'DuckDuckGo';
export const SEARCH_ECOSIA = 'Ecosia Search';
export const SEARCH_EKORU = 'ekoru';
export const SEARCH_GIBIRU = 'Gibiru.com';
export const SEARCH_GOOGLE = 'Google';
export const SEARCH_ONESEARCH = 'OneSearch';
export const SEARCH_PRIVADO = 'Privado';
export const SEARCH_QWANT = 'Qwant';
export const SEARCH_ENCRYPT = 'TODO: Add Search Encrypt browser.search.get() string';
export const SEARCH_TAILCAT = 'Tailcat';
export const SEARCH_OTHER = t('ghostery_dawn_onboarding_other');
