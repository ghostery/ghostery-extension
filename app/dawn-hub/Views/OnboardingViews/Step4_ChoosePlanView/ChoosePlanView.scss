/**
 * WelcomeView Sass
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2021 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

$medium-large-breakpoint: 1118px; // Break when 3 cards on the screen overflow to next line

.ChoosePlanView {
	margin: 45px auto 0 auto;
	padding-bottom: 20px;
}
.ChoosePlanView__relativeContainer {
	width: 724px;
	margin: auto;
	position: relative;
}
.ChoosePlanView__caret.left {
	margin: 7px auto 0 auto;
    height: 10px;
    width: 10px;
    border-left: 2px solid $tundora;
    border-top: 2px solid $tundora;
	cursor: pointer;
	transform: rotate(-45deg);
}
.ChoosePlanView__backContainer {
	position: absolute;
	display: flex;
	width: 68px;
	margin-top: 7px;
	.ChoosePlanView__back {
		margin-top: 8px;
		font-size: 16px;
		color: $tundora;
		text-decoration: underline;
	}
}
.ChoosePlanView__yourPrivacyPlan {
	margin: auto;
	font-size: 24px;
	display: flex;
	justify-content: center;
  	font-weight: 500;
}
.ChoosePlanView__subtitle {
	margin-top: 12px;
	display: flex;
	justify-content: center;
	font-size: 18px;
}
.ChoosePlanView__searchPromoContainer {
	width: 354px;
	height: 381px;
	margin: 48px auto 0 auto;
	border-radius: 4px;
	border: solid 4px $ghosty-blue;
}
.ChoosePlanView__searchLogo {
	height: 36px;
	width: 166px;
	margin: 65px auto 0 auto;
	background-image: url('/app/images/hub/ChooseDefaultSearchView/search-engine-logo-ghostery.svg');
}
.ChoosePlanView__adFree {
	margin: 16px auto 0 auto;
	color: #b8860b;
	font-size: 18px;
	line-height: 32px;
	font-weight: 600;
	width: 220px;
	text-align: center;
}
.ChoosePlanView__adFreePromo {
	margin: auto;
	font-size: 18px;
	line-height: 32px;
	text-align: center;
}
.ChoosePlanView__adFreePromoDescription {
	margin: 16px auto 0 auto;
	height: 88px;
	width: 250px;
	font-size: 16px;
	line-height: 1.38;
	text-align: center;
	color: $tundora;
}
.ChoosePlanView__searchCTAButton {
	display: flex;
	justify-content: center;
	margin: 48px auto 0 auto;
	height: 44px;
	width: 162px;
	padding: 7.7px 14px;
	line-height: 22px;
	background: linear-gradient(
		45deg,
		#ff7e74 50%,
		$ghosty-blue
	);
	background-size: 200% 100%;
	background-position: 100% 50%;
	transition: 0.25s all;
	border: none;
	color: $white;
	font-size: 14.1px;
	font-weight: 700;
	border-radius: 3.5px;
	text-align: center;
	line-height: 2.05;
	cursor: pointer;

	&:hover {
		background-position: 0% 50%;
		transition: 0.25s all;
		color: $white;
	}
	&:focus {
		color: $white;
	}
}
.ChoosePlanView__premiumCTAButton {
	display: flex;
	justify-content: center;
	margin: 48px auto 0 auto;
	height: 44px;
	width: 162px;
	padding: 7.7px 14px;
	line-height: 22px;
	background: linear-gradient(
			45deg,
 			#6c097a 50%,
 			#03a9ec
	);
	background-size: 200% 100%;
	background-position: 100% 50%;
	transition: 0.25s all;
	border: none;
	color: $white;
	font-size: 14.1px;
	font-weight: 700;
	border-radius: 3.5px;
	text-align: center;
	line-height: 2.05;
	cursor: pointer;

	&:hover {
		color: $white;
		background-position: 0% 50%;
		transition: 0.25s all;
	}
}
.ChoosePlanView__seeAllPlans {
	margin-top: 48px;
	display: flex;
	justify-content: center;
	color: $ghosty-blue;
	font-size: 16px;
	text-decoration: underline;
	cursor: pointer;
}
.ChoosePlanView__plansContainer {
	margin-top: 20px;
	padding-top: 20px;

	.ChoosePlanView__cardOuter {
		display: flex;
		flex-direction: column;
		justify-content: center;
		margin: 0 15px;
		min-height: 670px;
	}
}
.ChoosePlanView__orCardContainer {
	display: flex;
}
.ChoosePlanView__or {
	display: flex;
	align-items: center;
	justify-content: center;
	padding: 0 66px;
	height: 180px;
	font-size: 24px;
	color: $tundora;
}
.ChoosePlanView__ctaButtonContainer {
	display: flex;
	justify-content: center;
}
.ChoosePlanView__keepOrUpgradeContainer {
	margin: 40px auto;
}
.ChoosePlanView__arrow {
	margin: 15px auto 0 auto;
	height: 12px;
	width: 12px;
	border-left: 2px solid $ghosty-blue;
	border-top: 2px solid $ghosty-blue;
	cursor: pointer;

	&.down {
		transform: rotate(45deg);
	}
	&.up {
		transform: rotate(225deg);
	}
}
.ChoosePlanView__card {
	height: 100%;
	box-shadow: 0 0 7px 0 rgba(0, 0, 0, 0.5);
	margin-bottom: rem-calc(40);
	width: 250px;
	margin-left: auto;
	margin-right: auto;
	text-align: center;
	padding-bottom: rem-calc(40);
	border: 4px solid white;

	&:hover, &.checked {
		border: 4px solid $ghosty-blue;
		.ghostery-free-image-container,
		.ghostery-plus-image-container,
		.ghostery-premium-image-background {
			width: 244px;
			margin-left: -1px;
		}
	}
	h2 {
		color: $tundora;
		font-family: "Roboto Condensed";
		font-size: 18px;
		font-weight: 400;
		margin-top: 22px;
		margin-bottom: 6px;
	}
	&-image-top {
		margin-top: -76px;
		width: 100%;
	}
	&-sub-header {
		margin-bottom: rem-calc(15);
	}
	.ghostery-free-image-container {
		margin: -3px 0 0 -4px;
		width: 250px;
		height: 85px;
		background-color: rgba(0, 174, 240, .25);
		display: flex;
		justify-content: center;
		align-items: center;

		.ghostery-free-image {
			width: 55px;
			height: 65px;
			background-image: url('/app/images/hub/upgrade/ghostery-basic-card.svg');
			background-repeat: no-repeat;
			background-position: center;
		}
	}
	.ghostery-plus-image-container {
			margin: -3px 0 0 -4px;
			width: 250px;
			height: 85px;
			background-image: linear-gradient(to right, rgba(240,174,133,.25), rgba(241,216,158,.25));
		.ghostery-plus-image {
			margin: auto;
			width: 225px;
			height: 87px;
			background-image: url('/app/images/hub/upgrade/ghostery-plus-card.svg')
		}
	}
	.ghostery-premium-image-container {
		margin-left: -4px;
		position: relative;
	}
	.ghostery-premium-image {
		position: absolute;
		margin-top: -79px;
		margin-left: 9px;
		width: 233px;
		height: 161px;
		background-image: url('/app/images/hub/upgrade/ghostery-premium-card.svg');
	}
	.ghostery-premium-image-background {
		margin: -3px 0 0 -4px;
		width: 250px;
		height: 85px;
		background-image: linear-gradient(to right, rgba(113,1,116,.25), rgba(90,37,142,.25) 18%, rgba(42,110,194,.25) 57%, rgba(12,156,227,.25) 86%, rgba(0,174,240,.25));
	}
	.card-sub-header {
		font-size: 14px;
	}
}
.ChoosePlanView__price {
	min-height: 85px;
	margin-bottom: rem-calc(20);
	font-size: 36px;
	&-blue {
		color: $price-blue;
	}
	&-blue-alt {
		color: $blue-alt;
	}
	&-gold {
		color: $price-gold;
	}
	&-purple {
		color: $price-purple;
	}
	p {
		margin-bottom: 0;
		@include breakpoint($medium-large-breakpoint down) {
			&.sub-text:first-child {
				margin-top: 4px;
			}
		}
		&.sub-text {
			font-size: 12px;
			font-weight: 500;
		}
	}
}
.ChoosePlanView__valuePropList {
	margin: auto;
	width: 174px;
}
.ChoosePlanView__cardSubCopy {
	display: flex;
	justify-content: flex-start;
	margin-bottom: 5px;
	text-align: left;
}
.check {
	display: inline-block;
	width: 20px;
	height: 20px;
	background-repeat: no-repeat;
	margin-top: 3px;
	padding-right: 20px;
	&.blue {
		background-image: buildCheckIcon($price-blue);
	}
	&.yellow {
		background-image: buildCheckIcon($price-gold);
	}
	&.purple {
		background-image: buildCheckIcon($price-purple);
	}
}
.ChoosePlanView__radioButtonContainer {
	padding: 11px;
	margin-bottom: 84px;
	display: flex;
	justify-content: center;
}
