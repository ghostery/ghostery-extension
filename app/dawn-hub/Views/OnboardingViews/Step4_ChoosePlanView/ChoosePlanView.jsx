/**
 * Dawn Hub onboarding flow Choose Plan View Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2021 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React, { Fragment } from 'react';
import { NavLink } from 'react-router-dom';
import ClassNames from 'classnames';
import PropTypes from 'prop-types';
import RadioButton from '../../../../shared-components/RadioButton';
import globals from '../../../../../src/classes/Globals';
import { BASIC, PLUS, PREMIUM } from '../../../../hub/Views/UpgradePlanView/UpgradePlanViewConstants';
import {
	CHOOSE_PLAN,
	ONBOARDING,
	FREE_USER_NO_TRIAL,
	FREE_USER_PLUS_TRIAL,
	FREE_USER_PLUS_SUBSCRIPTION,
	FREE_USER_PREMIUM_SUBSCRIPTION,
	PLUS_SUBSCRIBER_KEEP_SUBSCRIPTION,
	PLUS_SUBSCRIBER_PREMIUM_SUBSCRIPTION,
	PREMIUM_SUBSCRIBER_KEEP_SUBSCRIPTION
} from '../../OnboardingView/OnboardingConstants';
import { SEARCH_GHOSTERY } from '../Step3_ChooseDefaultSearchView/ChooseDefaultSearchConstants';

const glowFreeTrialLink = `${globals.GLOWSTERY_BASE_URL}/account?utm_source=dawn&utm_medium=introhub&utm_campaign=onboarding`;
const plusCheckoutLink = `${globals.CHECKOUT_BASE_URL}/plus?utm_source=dawn&utm_medium=introhub&utm_campaign=onboarding`;
const premiumCheckoutLink = `${globals.CHECKOUT_BASE_URL}/premium?utm_source=dawn&utm_medium=introhub&utm_campaign=onboarding`;

const searchPromo = () => (
	<div className="ChoosePlanView__searchPromoContainer">
		<div className="ChoosePlanView__searchLogo" />
		<div className="ChoosePlanView__adFree">{ t('ghostery_dawn_onboarding_ad_free_with_ghostery_plus_subscription') }</div>
		<div className="ChoosePlanView__adFreePromo">
			{ `(${t('ghostery_dawn_onboarding_ad_free_promo')})` }
		</div>
		<div className="ChoosePlanView__adFreePromoDescription">{ t('ghostery_dawn_onboarding_ad_free_promo_description') }</div>
	</div>
);

const basicCard = (checked, handleClick) => {
	const cardClassNames = ClassNames('ChoosePlanView__card basic', {
		checked
	});
	return (
		<div className="ChoosePlanView__cardOuter">
			<div className={cardClassNames} onClick={handleClick} data-equalizer-watch>
				<div className="ChoosePlanView__radioButtonContainer">
					<RadioButton checked={checked} handleClick={handleClick} altDesign />
				</div>
				<div className="card-header-background-free" />
				<div className="ghostery-free-image-container">
					<div className="ghostery-free-image text-center" title="Ghostery Free" alt="Ghostery Free" />
				</div>
				<h2>Ghostery</h2>
				<div className="ChoosePlanView__price">
					<p className="ChoosePlanView__price-blue-alt font-size-36">{t('hub_upgrade_plan_free')}</p>
				</div>
				<p className="card-sub-header"><strong>{t('hub_upgrade_basic_protection')}</strong></p>
				<div className="ChoosePlanView__valuePropList basic">
					<div className="ChoosePlanView__cardSubCopy">
						<span className="check blue" />
						{t('ghostery_dawn_onboarding_private_search')}
					</div>
					<div className="ChoosePlanView__cardSubCopy">
						<span className="check blue" />
						{t('ghostery_dawn_onboarding_tracker_protection')}
					</div>
					<div className="ChoosePlanView__cardSubCopy">
						<span className="check blue" />
						{t('ghostery_dawn_onboarding_speedy_page_loads')}
					</div>
					<div className="ChoosePlanView__cardSubCopy">
						<span className="check blue" />
						{t('ghostery_dawn_onboarding_intelligence_technology')}
					</div>
				</div>
			</div>
		</div>
	);
};

class ChoosePlanView extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			selectedPlan: '',
			expanded: false
		};
		// User object doesn't get populated immediately, let's delay the first render
		setTimeout(this.setDefaultPlan, 200);
	}

	setDefaultPlan = () => {
		const { user } = this.props;
		const isPlus = (user && user.plusAccess) || false;
		const isPremium = (user && user.premiumAccess) || false;

		if (isPremium) {
			this.selectPremiumPlan();
			return;
		}
		if (isPlus) {
			this.selectPlusPlan();
			return;
		}
		this.selectBasicPlan();
	}

	isBasicPlanChecked = () => {
		const { selectedPlan } = this.state;
		return (selectedPlan === BASIC);
	};

	isPlusPlanChecked = () => {
		const { selectedPlan } = this.state;
		return (selectedPlan === PLUS);
	};

	isPremiumPlanChecked = () => {
		const { selectedPlan } = this.state;
		return (selectedPlan === PREMIUM);
	};

	selectBasicPlan = () => this.setState({ selectedPlan: BASIC });

	selectPlusPlan = () => this.setState({ selectedPlan: PLUS });

	selectPremiumPlan = () => this.setState({ selectedPlan: PREMIUM });

	toggleSection = () => {
		const { expanded } = this.state;
		if (expanded) {
			this.setState({ expanded: !expanded });
		} else {
			this.setState({ expanded: !expanded });
		}
	};

	renderTitleText = () => {
		const { user } = this.props;
		const isPlus = (user && user.plusAccess) || false;
		const isPremium = (user && user.premiumAccess) || false;

		if (isPremium) return t('ghostery_dawn_onboarding_already_premium_subscriber');
		if (isPlus) return t('ghostery_dawn_onboarding_already_plus_subscriber');
		return t('ghostery_dawn_onboarding_your_privacy_plan');
	};

	renderSubtitleText = (selectedGhosteryGlow) => {
		const { user } = this.props;
		const isPlus = (user && user.plusAccess) || false;
		const isPremium = (user && user.premiumAccess) || false;

		if (selectedGhosteryGlow) return t('ghostery_dawn_onboarding_based_on_your_privacy_preferences');
		if (isPremium) return '';
		if (isPlus) return t('ghostery_dawn_onboarding_keep_your_current_plan_or_upgrade');
		return t('ghostery_dawn_onboarding_choose_an_option');
	};

	setSetupStepAndMoveToSuccessView = (dawn_setup_number) => {
		const { actions, history } = this.props;
		const { setSetupStep } = actions;
		setSetupStep({ setup_step: CHOOSE_PLAN, dawn_setup_number, origin: ONBOARDING });
		history.push('/onboarding/5');
	}

	plusCard = (checked, handleClick, showCTAButton = false) => {
		const { actions } = this.props;
		const { setSetupStep } = actions;
		const cardClassNames = ClassNames('ChoosePlanView__card plus', {
			checked
		});
		return (
			<Fragment>
				<div className="ChoosePlanView__cardOuter">
					<div className={cardClassNames} onClick={handleClick} data-equalizer-watch>
						<div className="ChoosePlanView__radioButtonContainer">
							<RadioButton checked={checked} handleClick={handleClick} altDesign />
						</div>
						<div className="ghostery-plus-image-container">
							<div className="ghostery-plus-image" title="Ghostery Plus" alt="Ghostery Plus" />
						</div>
						<h2>Ghostery Plus</h2>
						<div className="ChoosePlanView__price">
							<Fragment>
								<p className="ChoosePlanView__price-gold font-size-36">$4.99</p>
								<p className="ChoosePlanView__price-gold sub-text font-size-12">{t('per_month')}</p>
							</Fragment>
						</div>
						<p className="card-sub-header"><strong>{t('hub_upgrade_additional_protection')}</strong></p>
						<div className="ChoosePlanView__valuePropList">
							<div className="ChoosePlanView__cardSubCopy">
								<span className="check blue" />
								{t('ghostery_dawn_onboarding_private_search')}
							</div>
							<div className="ChoosePlanView__cardSubCopy">
								<span className="check blue" />
								{t('ghostery_dawn_onboarding_tracker_protection')}
							</div>
							<div className="ChoosePlanView__cardSubCopy">
								<span className="check blue" />
								{t('ghostery_dawn_onboarding_speedy_page_loads')}
							</div>
							<div className="ChoosePlanView__cardSubCopy">
								<span className="check blue" />
								{t('ghostery_dawn_onboarding_intelligence_technology')}
							</div>
							<div className="ChoosePlanView__cardSubCopy">
								<span className="check blue" />
								{t('ghostery_dawn_onboarding_ad_free')}
							</div>
							<div className="ChoosePlanView__cardSubCopy">
								<span className="check blue" />
								{t('ghostery_dawn_onboarding_supports_ghosterys_mission')}
							</div>
						</div>
					</div>
				</div>
				{showCTAButton && (
					<NavLink className="ChoosePlanView__searchCTAButton" to="/onboarding/5" onClick={() => setSetupStep({ setup_step: CHOOSE_PLAN, dawn_setup_number: PLUS_SUBSCRIBER_KEEP_SUBSCRIPTION, origin: ONBOARDING })}>
						<span>{t('ghostery_dawn_onboarding_keep')}</span>
					</NavLink>
				)}
			</Fragment>
		);
	};

	premiumCard = (checked, handleClick, showCTAButton = false) => {
		const cardClassNames = ClassNames('ChoosePlanView__card premium', {
			checked
		});
		return (
			<Fragment>
				<div className="ChoosePlanView__cardOuter">
					<div className={cardClassNames} onClick={handleClick} data-equalizer-watch>
						<div className="ChoosePlanView__radioButtonContainer">
							<RadioButton checked={checked} handleClick={handleClick} altDesign />
						</div>
						<div className="ghostery-premium-image-container">
							<div className="ghostery-premium-image card-image-top" title="Ghostery Premium" alt="Ghostery Premium" />
						</div>
						<div className="ghostery-premium-image-background" />
						<h2>Ghostery Premium</h2>
						<div className="ChoosePlanView__price">
							<Fragment>
								<p className="ChoosePlanView__price-purple font-size-36">$11.99</p>
								<p className="ChoosePlanView__price-purple sub-text font-size-12">{t('per_month')}</p>
							</Fragment>
						</div>
						<p className="card-sub-header"><strong>{t('hub_upgrade_maximum_protection')}</strong></p>
						<div className="ChoosePlanView__valuePropList">
							<div className="ChoosePlanView__cardSubCopy">
								<span className="check blue" />
								{t('ghostery_dawn_onboarding_private_search')}
							</div>
							<div className="ChoosePlanView__cardSubCopy">
								<span className="check blue" />
								{t('ghostery_dawn_onboarding_tracker_protection')}
							</div>
							<div className="ChoosePlanView__cardSubCopy">
								<span className="check blue" />
								{t('ghostery_dawn_onboarding_speedy_page_loads')}
							</div>
							<div className="ChoosePlanView__cardSubCopy">
								<span className="check blue" />
								{t('ghostery_dawn_onboarding_intelligence_technology')}
							</div>
							<div className="ChoosePlanView__cardSubCopy">
								<span className="check blue" />
								{t('ghostery_dawn_onboarding_ad_free')}
							</div>
							<div className="ChoosePlanView__cardSubCopy">
								<span className="check blue" />
								{t('ghostery_dawn_onboarding_supports_ghosterys_mission')}
							</div>
							<div className="ChoosePlanView__cardSubCopy">
								<span className="check blue" />
								VPN
							</div>
							<div className="ChoosePlanView__cardSubCopy">
								<span className="check blue" />
								{t('ghostery_dawn_onboarding_unlimited_bandwidth')}
							</div>
						</div>
					</div>
				</div>
				{showCTAButton && (
					<a className="ChoosePlanView__premiumCTAButton" href={premiumCheckoutLink} target="_blank" rel="noreferrer" onClick={() => this.setSetupStepAndMoveToSuccessView(PLUS_SUBSCRIBER_PREMIUM_SUBSCRIPTION)}>{t('ghostery_dawn_onboarding_upgrade')}</a>
				)}
			</Fragment>
		);
	};

	render() {
		const {
			actions,
			defaultSearch,
			user,
		} = this.props;
		const { setSetupStep } = actions;
		const { expanded, selectedPlan } = this.state;

		const isBasic = !user || (user && !user.plusAccess && !user.premiumAccess);
		const isPlus = (user && user.plusAccess && !user.premiumAccess) || false;
		const isPremium = (user && user.premiumAccess) || false;

		const arrowClassNames = ClassNames('ChoosePlanView__arrow', {
			up: !expanded,
			down: expanded
		});

		const selectedGhosteryGlow = (defaultSearch === SEARCH_GHOSTERY);

		return (
			<Fragment>
				<div className="ChoosePlanView__relativeContainer">
					<div className="ChoosePlanView__backContainer">
						<span className="ChoosePlanView__caret left" />
						<NavLink to="/onboarding/3">
							<span className="ChoosePlanView__back">{t('ghostery_dawn_onboarding_back')}</span>
						</NavLink>
					</div>
				</div>
				<div className="ChoosePlanView">
					<div className="ChoosePlanView__yourPrivacyPlan">{this.renderTitleText()}</div>
					<div className="ChoosePlanView__subtitle">{this.renderSubtitleText(selectedGhosteryGlow)}</div>
					{selectedGhosteryGlow && isBasic && (
						<Fragment>
							{searchPromo()}
							<div className="ChoosePlanView__searchCTAButtonContainer">
								<a className="ChoosePlanView__searchCTAButton" href={glowFreeTrialLink} target="_blank" rel="noreferrer" onClick={() => this.setSetupStepAndMoveToSuccessView(FREE_USER_PLUS_TRIAL)}>{t('ghostery_dawn_onboarding_start_trial')}</a>
							</div>
							<div className="ChoosePlanView__seeAllPlans" onClick={this.toggleSection}>{t('ghostery_dawn_onboarding_see_all_plans')}</div>
							<div className={arrowClassNames} onClick={this.toggleSection} />
						</Fragment>
					)}
					{((isBasic && !selectedGhosteryGlow) || expanded || isPlus || isPremium) && (
						<div>
							{(isPlus) ? (
								<div className="ChoosePlanView__keepOrUpgradeContainer row align-center align-middle">
									<div className="small-12 medium-12 large-4">
										{this.plusCard(this.isPlusPlanChecked(), this.selectPlusPlan, isPlus)}
									</div>
									<div className="ChoosePlanView__or small-12 large-2">{t('ghostery_dawn_onboarding_or')}</div>
									<div className="small-12 medium-12 large-4">
										{this.premiumCard(this.isPremiumPlanChecked(), this.selectPremiumPlan, isPlus)}
									</div>
								</div>
							) : (
								<div className="ChoosePlanView__plansContainer row align-spaced">
									{isBasic && (
										basicCard(this.isBasicPlanChecked(), this.selectBasicPlan)
									)}
									{!isPremium && (
										<Fragment>
											{this.plusCard(this.isPlusPlanChecked(), this.selectPlusPlan)}
										</Fragment>
									)}
									{this.premiumCard(this.isPremiumPlanChecked(), this.selectPremiumPlan)}
								</div>
							)}
							{(isBasic && (
								<div className="ChoosePlanView__ctaButtonContainer">
									{(selectedPlan === BASIC) && (
										<NavLink className="ChoosePlanView__searchCTAButton" to="/onboarding/5" onClick={() => setSetupStep({ setup_step: CHOOSE_PLAN, dawn_setup_number: FREE_USER_NO_TRIAL, origin: ONBOARDING })}>
											<span>{t('next')}</span>
										</NavLink>
									)}
									{selectedPlan === PLUS && (
										<a className="ChoosePlanView__searchCTAButton" onClick={() => this.setSetupStepAndMoveToSuccessView(FREE_USER_PLUS_SUBSCRIPTION)} href={plusCheckoutLink} target="_blank" rel="noreferrer">{t('next')}</a>
									)}
									{selectedPlan === PREMIUM && (
										<a className="ChoosePlanView__searchCTAButton" onClick={() => this.setSetupStepAndMoveToSuccessView(FREE_USER_PREMIUM_SUBSCRIPTION)} href={premiumCheckoutLink} target="_blank" rel="noreferrer">{t('next')}</a>
									)}
								</div>
							))}
							{isPremium && (
								<NavLink className="ChoosePlanView__searchCTAButton" to="/onboarding/5" onClick={() => setSetupStep({ setup_step: CHOOSE_PLAN, dawn_setup_number: PREMIUM_SUBSCRIBER_KEEP_SUBSCRIPTION, origin: ONBOARDING })}>
									<span>{t('next')}</span>
								</NavLink>
							)}
						</div>
					)}
				</div>
			</Fragment>
		);
	}
}

// PropTypes ensure we pass required props of the correct type
ChoosePlanView.propTypes = {
	user: PropTypes.shape({
		plusAccess: PropTypes.bool,
		premiumAccess: PropTypes.bool,
	}),
	defaultSearch: PropTypes.string.isRequired,
};

// Default props used in the Plus View
ChoosePlanView.defaultProps = {
	user: {
		plusAccess: false,
		premiumAccess: false,
	},
};

export default ChoosePlanView;
