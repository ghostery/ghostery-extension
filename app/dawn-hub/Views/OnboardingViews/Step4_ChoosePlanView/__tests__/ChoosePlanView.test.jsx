/**
 * ChoosePlanView Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2021 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router';
import ChoosePlanView from '../ChoosePlanView';
import { SEARCH_GHOSTERY } from '../../Step3_ChooseDefaultSearchView/ChooseDefaultSearchConstants';

const noop = () => {};

describe('app/dawn-hub/Views/OnboardingViews/Step4_ChoosePlanView/ChoosePlanView.test.jsx', () => {
	const initialState = {
		user: null,
		selectedGhosteryGlow: true,
		actions: {
			setSetupStep: noop
		},
		defaultSearch: SEARCH_GHOSTERY,
	};
	describe('Snapshot tests with react-test-renderer', () => {
		test('ChoosePlanView is rendered correctly', () => {
			const component = renderer.create(
				<MemoryRouter>
					<ChoosePlanView  {...initialState} />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('Shallow snapshot tests rendered with Enzyme', () => {
		test('ChoosePlanView View with user not logged in', () => {
			const component = shallow(<ChoosePlanView {...initialState} />);

			const instance = component.instance();

			instance.selectBasicPlan();
			expect(component.state('selectedPlan')).toBe('BASIC');

			instance.selectPlusPlan();
			expect(component.state('selectedPlan')).toBe('PLUS');

			instance.selectPremiumPlan();
			expect(component.state('selectedPlan')).toBe('PREMIUM');

			expect(component).toMatchSnapshot();
		});

		test('ChoosePlanView View with basic user logged in', () => {
			const basicUserState = {
				...initialState,
				user: {
					plusAccess: false,
					premiumAccess: false
				}
			};

			const component = shallow(<ChoosePlanView {...basicUserState} />);

			expect(component).toMatchSnapshot();
		});

		test('ChoosePlanView View with plus user logged in', () => {
			const plusUserState = {
				...initialState,
				user: {
					plusAccess: true,
					premiumAccess: false
				},
			};

			const component = shallow(<ChoosePlanView {...plusUserState} />);

			expect(component).toMatchSnapshot();
		});

		test('ChoosePlanView View with premium user logged in', () => {
			const premiumUserState = {
				...initialState,
				user: {
					plusAccess: true,
					premiumAccess: true
				}
			};

			const component = shallow(<ChoosePlanView {...premiumUserState } />);

			expect(component).toMatchSnapshot();
		});
	});
});
