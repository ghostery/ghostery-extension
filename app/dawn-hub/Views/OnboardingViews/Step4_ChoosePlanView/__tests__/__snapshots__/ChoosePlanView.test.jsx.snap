// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/dawn-hub/Views/OnboardingViews/Step4_ChoosePlanView/ChoosePlanView.test.jsx Shallow snapshot tests rendered with Enzyme ChoosePlanView View with basic user logged in 1`] = `ShallowWrapper {}`;

exports[`app/dawn-hub/Views/OnboardingViews/Step4_ChoosePlanView/ChoosePlanView.test.jsx Shallow snapshot tests rendered with Enzyme ChoosePlanView View with plus user logged in 1`] = `ShallowWrapper {}`;

exports[`app/dawn-hub/Views/OnboardingViews/Step4_ChoosePlanView/ChoosePlanView.test.jsx Shallow snapshot tests rendered with Enzyme ChoosePlanView View with premium user logged in 1`] = `ShallowWrapper {}`;

exports[`app/dawn-hub/Views/OnboardingViews/Step4_ChoosePlanView/ChoosePlanView.test.jsx Shallow snapshot tests rendered with Enzyme ChoosePlanView View with user not logged in 1`] = `ShallowWrapper {}`;

exports[`app/dawn-hub/Views/OnboardingViews/Step4_ChoosePlanView/ChoosePlanView.test.jsx Snapshot tests with react-test-renderer ChoosePlanView is rendered correctly 1`] = `
Array [
  <div
    className="ChoosePlanView__relativeContainer"
  >
    <div
      className="ChoosePlanView__backContainer"
    >
      <span
        className="ChoosePlanView__caret left"
      />
      <a
        aria-current={null}
        href="/onboarding/3"
        onClick={[Function]}
      >
        <span
          className="ChoosePlanView__back"
        >
          ghostery_dawn_onboarding_back
        </span>
      </a>
    </div>
  </div>,
  <div
    className="ChoosePlanView"
  >
    <div
      className="ChoosePlanView__yourPrivacyPlan"
    >
      ghostery_dawn_onboarding_your_privacy_plan
    </div>
    <div
      className="ChoosePlanView__subtitle"
    >
      ghostery_dawn_onboarding_based_on_your_privacy_preferences
    </div>
    <div
      className="ChoosePlanView__searchPromoContainer"
    >
      <div
        className="ChoosePlanView__searchLogo"
      />
      <div
        className="ChoosePlanView__adFree"
      >
        ghostery_dawn_onboarding_ad_free_with_ghostery_plus_subscription
      </div>
      <div
        className="ChoosePlanView__adFreePromo"
      >
        (ghostery_dawn_onboarding_ad_free_promo)
      </div>
      <div
        className="ChoosePlanView__adFreePromoDescription"
      >
        ghostery_dawn_onboarding_ad_free_promo_description
      </div>
    </div>
    <div
      className="ChoosePlanView__searchCTAButtonContainer"
    >
      <a
        className="ChoosePlanView__searchCTAButton"
        href="https://staging.glowstery.com/account?utm_source=dawn&utm_medium=introhub&utm_campaign=onboarding"
        onClick={[Function]}
        rel="noreferrer"
        target="_blank"
      >
        ghostery_dawn_onboarding_start_trial
      </a>
    </div>
    <div
      className="ChoosePlanView__seeAllPlans"
      onClick={[Function]}
    >
      ghostery_dawn_onboarding_see_all_plans
    </div>
    <div
      className="ChoosePlanView__arrow up"
      onClick={[Function]}
    />
  </div>,
]
`;
